src/components/popover/popover.ts

```diff
@@ -1,112 +1,294 @@
+  onPopoverHidden?: () => {};
 }
 
-export class Popover extends Component<Props, SpreadsheetChildEnv> {
+css/* scss */ `
+  .o-popover {
+    position: absolute;
+    z-index: ${ComponentsImportance.Popover};
+    overflow: auto;
+    box-shadow: 1px 2px 5px 2px rgb(51 51 51 / 15%);
+    width: fit-content;
+    height: fit-content;
+  }
+`;
+
+export class Popover extends Component<PopoverProps, SpreadsheetChildEnv> {
   static template = "o-spreadsheet-Popover";
   static defaultProps = {
-    flipHorizontalOffset: 0,
-    flipVerticalOffset: 0,
+    positioning: "BottomLeft",
     verticalOffset: 0,
-    marginTop: 0,
     onMouseWheel: () => {},
+    onPopoverMove: () => {},
+    onPopoverHidden: () => {},
   };
 
-  private spreadsheetPosition = useSpreadsheetPosition();
-
-  get style() {
-    // the props's position is expressed relative to the "body" element
-    // but we teleport the element in ".o-spreadsheet" to keep everything
-    // within our control and to avoid leaking into external DOM
-    const horizontalPosition = `left:${this.horizontalPosition() - this.spreadsheetPosition.x}`;
-    const verticalPosition = `top:${this.verticalPosition() - this.spreadsheetPosition.y}`;
-    const height = `max-height:${
-      this.viewportDimension.height - BOTTOMBAR_HEIGHT - SCROLLBAR_WIDTH
-    }`;
-    return `
-      position: absolute;
-      z-index: ${ComponentsImportance.Popover};
-      ${verticalPosition}px;
-      ${horizontalPosition}px;
-      ${height}px;
-      width:${this.props.childWidth}px;
-      overflow-y: auto;
-      overflow-x: hidden;
-      box-shadow: 1px 2px 5px 2px rgb(51 51 51 / 15%);
-    `;
-  }
-
-  private get viewportDimension(): DOMDimension {
-    return this.env.model.getters.getSheetViewDimensionWithHeaders();
-  }
-
-  private get shouldRenderRight(): boolean {
-    const { x } = this.props.position;
-    return x + this.props.childWidth < this.viewportDimension.width;
-  }
-
-  private get shouldRenderBottom(): boolean {
-    const { y } = this.props.position;
+  private popoverRef = useRef("popover");
+  private currentPosition: PopoverPosition | undefined = undefined;
+  private currentVisibility: Visibility | undefined = undefined;
+
+  setup() {
+    // useEffect occurs after the DOM is created and the element width/height are computed, but before
+    // the element in rendered, so we can still set its position
+    useEffect(() => {
+      this.resetPopoverElStyle();
+      const el = this.popoverRef.el!;
+
+      const propsMaxSize = { width: this.props.maxWidth, height: this.props.maxHeight };
+
+      const elDims = {
+        width: el.getBoundingClientRect().width,
+        height: el.getBoundingClientRect().height,
+      };
+
+      const spreadsheetRect = el.parentElement!.getBoundingClientRect();
+      const spreadsheetOffset = { x: spreadsheetRect.left, y: spreadsheetRect.top };
+
+      const containerRect = {
+        x: this.props.containerRect?.x || spreadsheetOffset.x,
```

Comment:
if `this.props.containerRect?.x` is `0`, it will take `spreadsheetOffset.x`.
Same below